cromshell

```diff
@@ -432,19 +518,29 @@ function abort()
 # List all jobs submitted in ${CROMWELL_SUBMISSIONS_FILE}
 function list()
 {
-  turtle
   local doColor=false
   local doUpdate=false
-
-  for arg in ${@} ; do 
-    case "$arg" in
-      -c) doColor=true
-         ;; 
-      -u) doUpdate=true
-         ;; 
-    esac
-  done
  
+  # Handle Arguments:
+  local OPTIND
+  while getopts "cu" opt ; do
+    case ${opt} in 
+      c)
+        doColor=true
+        ;;
+      u)
+        doUpdate=true
+        ;;
+      *)
+        invalidSubCommand list flag ${OPTARG}
+        ;; 
+    esac 
+  done
+  shift $((OPTIND-1)) 
```

Comment:
I don't really know how getopts works.  What's this shift doing after the opt parsing loop? Should it go within the opt parsing loop to loop over all the options?

Reply:
Also, where is `opt` defined?

Reply:
It's just a more sane option parser.  The only problem is that while it works well for single character flags, it doesn't work well for positional args or long args (e.g. `--help`), so it ended up being a little less useful for cromshell as it stands now.

That `shift` is shifting the arguments by the number of args that have been parsed by `getopt`.

`opt` is defined at the top of the while loop (line 526): `while getopts "cu" `**opt**` ; do`

Reply:
Oh, it's defined by getopts... that makes sense... Does getopt default to using `$@` as it's input?

Reply:
That's my understanding, which is how it can be used in functions (providing you mark `OPTIND` as local).